src/main/kotlin/coverosR3z/timerecording/api/ViewTimeAPI.kt

```diff
@@ -130,19 +150,20 @@ class ViewTimeAPI(private val sd: ServerData) {
     private fun renderTimeRows(
         te: Set<TimeEntry>,
         idBeingEdited: Int?,
-        projects: List<Project>
+        projects: List<Project>,
+        currentPeriod: TimePeriod
     ): String {
         return renderCreateTimeRow(projects) +
         te.sortedBy { it.id.value }.joinToString("") {
             if (it.id.value == idBeingEdited) {
-                renderEditRow(it, projects)
+                renderEditRow(it, projects, currentPeriod)
             } else {
-                renderReadOnlyRow(it)
+                renderReadOnlyRow(it, currentPeriod)
             }
         }
     }
 
-    private fun renderReadOnlyRow(it: TimeEntry): String {
+    private fun renderReadOnlyRow(it: TimeEntry, currentPeriod: TimePeriod): String {
```

Comment:
Now we pass around the current period to various parts of the page, since various actions may cause reloading of the page and we the ability to have the proper time period shown when that happens